### PR TITLE
[SPARK-24186][R][SQL]change reverse and concat to collection functions in R

### PR DESCRIPTION
**Pull request by huaxingao**
## What changes were proposed in this pull request?

reverse and concat are already in functions.R as column string functions. Since now these two functions are categorized as collection functions in scala and python, we will do the same in R.

## How was this patch tested?

Add test in test_sparkSQL.R

### CODE REVIEW

**Review by HyukjinKwon — APPROVED** [review]
@huaxingao, sorry for the conflict. LGTM except for the conflict cc @felixcheung too

**Comment by HyukjinKwon** [comment]
@huaxingao. feel free to reopen a PR if you get any problem. It's fine.

**Comment by viirya** [comment]
LGTM

File: R/pkg/R/functions.R [hunk 1]
```diff
@@ -2055,20 +2058,10 @@ setMethod("countDistinct",
 
 #' @details
 #' \code{concat}: Concatenates multiple input columns together into a single column.
-#' If all inputs are binary, concat returns an output as binary. Otherwise, it returns as string.
+#' The function works with strings, binary and compatible array columns.
```
Comment (felixcheung) [inline R/pkg/R/functions.R]: btw, what does "compatible array columns" mean?

**Review by felixcheung — APPROVED** [review]
LGTM

File: R/pkg/R/functions.R (hunk 1 above)
Comment (huaxingao) [reply R/pkg/R/functions.R]: @felixcheung 
I copied the sentence from scala doc. I guess it means "array columns that have data type compatible with string and binary", but I am not sure.

Comment (HyukjinKwon) [reply R/pkg/R/functions.R]: I think it basically mean the arrays of elements are compatibile and castable for a wider type. (e.g. array<int> array<long> -> array<long>)

**Comment by HyukjinKwon** [comment]
Merged to master.

**Comment by huaxingao** [comment]
Thanks a lot!! @HyukjinKwon @viirya @felixcheung

**Pull request closed without merging**